src/main/scala/com/microsoft/partnercatalyst/fortis/spark/TransformManager.scala

```
@@ -0,0 +1,256 @@
+
+  // Do not serialize these values. The transformContext would otherwise contain stale Broadcast
+  // instances, so we rebuild it on recovery from checkpoint.
+  @volatile @transient private var transformContext: TransformContext = null
+  @volatile @transient private var queueClient: QueueClient = null
+
+  def getOrUpdateContext(sparkContext: SparkContext): TransformContext = {
+    ensureInitialized(sparkContext)
+
+    writeLock.lock()
+    try {
+      // Grab delta from hand-off only if it's available *now*
+      val delta = Option(shouldUpdate.poll(0, TimeUnit.SECONDS))
+
+      // Update transform context with delta
+      delta.foreach(updateTransformContext(_, sparkContext))
+    }
+    finally {
+      writeLock.unlock()
+    }
+
+    transformContext
+  }
+
+  private def ensureInitialized(sparkContext: SparkContext): Unit = {
+    if (transformContext == null) {
+      // Blocking init of transform context and queue client used for non-blocking updates.
+      // Initialization has not yet occurred since construction/deserialization.
+      writeLock.lock()
+      try {
+        if (transformContext == null) {
+
+          // Fetch data synchronously
+          val siteSettings = configManager.fetchSiteSettings()
+          val langToWatchlist = configManager.fetchWatchlist()
+          val blacklist = configManager.fetchBlacklist()
+
+          var delta = deltaWithSettings(siteSettings)
+          delta = deltaWithWatchlist(langToWatchlist, delta)
+          delta = deltaWithBlacklist(blacklist, delta)
+
+          // Create transformContext and broadcast
+          updateTransformContext(delta, sparkContext)
+
+          // Start async update listener
+          startQueueClient()
+        }
+      }
+      finally {
+        writeLock.unlock()
+      }
+    }
+  }
+
+  private def startQueueClient(): Unit = {
+    queueClient = new QueueClient(
+      new ConnectionStringBuilder(
+        Properties.envOrElse("FORTIS_SERVICEBUS_NAMESPACE", ""),
```

Should probably fail early in case the required environment variables are not set.

---

My thought is that we should ditch the whole `Settings` thing and just hit env vars at time of usage. We can add early fails somewhere in the main. I'll add constants for these env var names for now.